Scripts/RemediationScripts/Remediate-SetStorageAccountMinReqTLSVersion.ps1

```diff
@@ -0,0 +1,959 @@
+        Write-Host "Account Type: $($context.Account.Type)"
+        Write-Host $([Constants]::SingleDashLine)
+    }
+
+    Write-Host "To Set minimal TLS version for Storage Accounts in a Subscription, Contributor or higher privileges on the Storage Accounts are required." -ForegroundColor $([Constants]::MessageType.Warning)
+    Write-Host $([Constants]::SingleDashLine)
+    Write-Host "[Step 2 of 4] Fetch all Storage Accounts"
+    Write-Host $([Constants]::SingleDashLine)
+    
+    $storageAccountResources = @()
+    $requiredMinTLSVersion = "TLS1_2"
+
+
+    # To keep track of remediated and skipped resources
+    $logRemediatedResources = @()
+    $logSkippedResources = @()
+
+    # Control Id
+    $controlIds = "Azure_Storage_DP_Use_Secure_TLS_Version_Trial"
+
+    if($AutoRemediation)
+    {
+        if(-not (Test-Path -Path $Path))
+        {
+        Write-Host "File containing failing controls details [$($Path)] not found. Skipping remediation..." -ForegroundColor $([Constants]::MessageType.Error)
+        Write-Host $([Constants]::DoubleDashLine)
+        return
+        }
+        Write-Host "Fetching all Storage Accounts failing for the [$($controlIds)] control from [$($Path)]..." -ForegroundColor $([Constants]::MessageType.Info)
+        Write-Host $([Constants]::SingleDashLine)
+        $controlForRemediation = Get-content -path $Path | ConvertFrom-Json
+        $controls = $controlForRemediation.ControlRemediationList
+        $resourceDetails = $controls | Where-Object { $controlIds -eq $_.ControlId };
+        $validResources = $resourceDetails.FailedResourceList | Where-Object {![String]::IsNullOrWhiteSpace($_.ResourceId)}
+
+        if(($resourceDetails | Measure-Object).Count -eq 0 -or ($validResources | Measure-Object).Count -eq 0)
+        {
+            Write-Host "No Storage Account(s) found in input json file for remediation." -ForegroundColor $([Constants]::MessageType.Error)
+            Write-Host $([Constants]::DoubleDashLine)
+            return
+        }  
+        $validResources | ForEach-Object { 
+            try
+            {
+                $name = $_.ResourceName
+                $resourceGroupName = $_.ResourceGroupName
+            
+                $resStorageAccount = Get-AzStorageAccount  -Name $name  -ResourceGroupName $resourceGroupName -ErrorAction SilentlyContinue
+                $storageAccountResources = $resStorageAccount | Select-Object @{N='StorageAccountName';E={$_.StorageAccountName}},
+                                                                            @{N='ResourceGroupName';E={$_.ResourceGroupName}},
+                                                                            @{N='PrimaryLocation';E={$_.PrimaryLocation}},
+                                                                            @{Name ='MinimalTlsVersion';Expression={ (Get-AzStorageAccount -Name $_.StorageAccountName -ResourceGroupName $_.ResourceGroupName).MinimumTlsVersion }}
```

Comment:
Why are we doing Get-AzStorageAccount twice?

Reply:
Replaced two calls with one. 
The parameter MinimalTlsVersion is coming when we explicitly get it using . operator. But not coming in as paramater when we run Get-AzStorageAccount